chef/cookbooks/hyperv/recipes/config_ceilometer.rb

```diff
@@ -0,0 +1,103 @@
+      db_connection = "mongodb://#{mongodb_servers.sort.join(',')}/ceilometer?replicaSet=#{node[:ceilometer][:ha][:mongodb][:replica_set][:name]}"
+    end
+  end
+
+  # if this is a cluster, but the replica set member attribute hasn't
+  # been set on any node (yet), we just fallback to using the first
+  # ceilometer-server node
+  if db_connection.nil?
+    db_hosts = search(:node, "roles:ceilometer-server")
+    db_host = db_hosts.first || node
+    ## This will fail on windows
+    mongodb_ip = Chef::Recipe::Barclamp::Inventory.get_network_by_type(db_host, "admin").address
+    db_connection = "mongodb://#{mongodb_ip}:#{db_host[:ceilometer][:mongodb][:port]}/ceilometer"
+  end
+else
+  db_settings = fetch_database_settings
+
+  include_recipe "database::client"
+  include_recipe "#{db_settings[:backend_name]}::client"
+  include_recipe "#{db_settings[:backend_name]}::python-client"
+
+  db_password = ''
+  # pickup password to database from ceilometer-server node
+  node_controllers = search(:node, "roles:ceilometer-server") || []
+  if node_controllers.length > 0
+    db_password = node_controllers[0][:ceilometer][:db][:password]
+  end
+
+  db_connection = "#{db_settings[:url_scheme]}://#{node[:ceilometer][:db][:user]}:#{db_password}@#{db_settings[:address]}/#{node[:ceilometer][:db][:database]}"
+end
+
+
+time_to_live = node[:ceilometer][:database][:time_to_live]
+if time_to_live > 0
+  # We store the value of time to live in days, but config file expects seconds
+  time_to_live = time_to_live * 3600 * 24
+end
+
+is_compute_agent = %w(ceilometer-agent-hyperv nova-multi-compute-hyperv).all?{|role| node.roles.include?(role)}
+
+dirs = [ node[:openstack][:ceilometer][:lock_path], node[:openstack][:ceilometer][:signing_dir] ]
+dirs.each do |dir|
+  directory dir do
+    action :create
+    recursive true
+  end
+end
+
+template "#{node[:openstack][:config].gsub(/\\/, "/")}/ceilometer.conf" do
+    source "ceilometer.conf.erb"
+    variables(
+      :debug => node[:ceilometer][:debug],
+      :verbose => node[:ceilometer][:verbose],
+      :rabbit_settings => fetch_rabbitmq_settings("ceilometer"),
+      :keystone_settings => keystone_settings,
+      :bind_host => bind_host,
+      :bind_port => bind_port,
+      :metering_secret => node[:ceilometer][:metering_secret],
+      :database_connection => db_connection,
+      :node_hostname => node['hostname'],
+      :hypervisor_inspector => "hyperv",
+      :libvirt_type => "",
+      :time_to_live => time_to_live,
+      :alarm_threshold_evaluation_interval => node[:ceilometer][:alarm_threshold_evaluation_interval],
+      :lock_path => node[:openstack][:ceilometer][:lock_path],
+      :log_dir =>  node[:openstack][:log],
+      :signing_dir => node[:openstack][:ceilometer][:signing_dir]
+    )
+    if is_compute_agent
+      notifies :restart, "service[nova-compute]"
+    end
+end
+
+template "#{node[:openstack][:config].gsub(/\\/, "/")}/pipeline.yaml" do
```

Comment:
No `# Chef 11.4 fails to notify if the path separator is windows like, ...` comment here?